File: Inventory-Management-System/src/main/java/com/brocode/entity/Batch.java
package com.brocode.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;
import lombok.experimental.SuperBuilder;

import java.util.ArrayList;
import java.util.List;

@Data
@Entity
@AllArgsConstructor
@NoArgsConstructor
@SuperBuilder
@EqualsAndHashCode(callSuper = true)
public class Batch extends BaseEntity{

    @ManyToOne
    @JoinColumn(name = "product_id")
    private Product product;

    @ManyToOne
    @JoinColumn(name = "order_id")
    private Order order;

    private Double orderPrice;

    @OneToMany(mappedBy = "batch", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<SaleItem> sales = new ArrayList<>();

    private Double stockLeft;

}

File: mobile-application/app/src/main/java/com/example/inventory/data/Batch.kt
package com.example.inventory.data

import java.time.LocalDate

data class BatchResponseDto(
    val orderId: Long,
    val productId: Long,
    val productName: String,
    val sales: List<SaleItemResponseDto>,
    val stockLeft: Double,
    val orderDate: LocalDate,
    val orderPrice: Double
): BaseResponseDto()